keps/sig-network/3015-node-level-topology/README.md

```diff
@@ -0,0 +1,502 @@
+the version skew should eventually get resolved.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: NodeLevelTopology
+  - Components depending on the feature gate:
+    - kube-apiserver
+    - kube-controller-manager
+    - kube-proxy
+
+###### Does enabling the feature change any default behavior?
+
+Potentially, depending on UNRESOLVED decisions above.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes; services that were using node-level topology will just revert
+back to ordinary `Cluster` semantics.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+It starts working again.
+
+###### Are there any tests for feature enablement/disablement?
+
+Not yet.
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Assuming we use the "Automatic" semantics, a rollout would affect the
+behavior of existing services (such as CoreDNS) that met the criteria
+for automatically getting node-local topology. Assuming the feature
+works correctly though, it would affect them in a positive way.
+
+If we used the "Manual" semantics then rolling out the feature should
+have no effect at all.
+
+###### What specific metrics should inform a rollback?
+
+There are no metrics that would inform anyone that the feature was
+failing (discussed more below).
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+No
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+Assuming the "Manual" model, they could check all services to see if
+they were annotated to use the feature. Assuming the "Automatic"
+model, there is no easy way to determine thisl
+
+###### How can someone using this feature know that it is working for their instance?
+
+They can know that it hasn't broken anything because their services
+won't be broken.
+
+There's no simple way to confirm that kube-proxy is optimizing traffic
+correctly based on this feature. The user would need to packet-sniff,
+or check individual endpoint logs, or look at the iptables rules
+generated by kube-proxy.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+Services using node-level topology should be as reliable as services
+_not_ using node-level topology, and should have lower latency.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [X] Other
+
+There are no existing metrics for determining the availability or
```

Comment:
"service" in the question refers to the feature, really. So in this case, any existing kube-proxy SLIs would apply. For example, programming time & latency, kube-proxy crashes, etc.